### PR TITLE
8.0 Fixed PXB-2715 validate decompression path

### PR DESCRIPTION
**Pull request by altmannmarcelo**
Problem:
While decompressing a backup, we pass filepath and dest_filepath to
system call without any validation.

Fix:
escape all special characters within the file name and quote the file
before invoking qpress command.

### CODE REVIEW

File: storage/innobase/xtrabackup/src/backup_copy.cc [hunk 1]
```diff
@@ -2570,10 +2570,16 @@ bool copy_back(int argc, char **argv) {
 
 bool decrypt_decompress_file(const char *filepath, uint thread_n) {
   std::stringstream cmd, message;
-  char *dest_filepath = strdup(filepath);
+  char buf[FN_LEN];
   bool needs_action = false;
 
-  cmd << "cat " << filepath;
+  if (escape_string_for_mysql(&my_charset_utf8mb4_general_ci, buf, 0, filepath,
+                              strlen(filepath)) == (size_t)-1) {
```
Comment (satya-bodapati) [inline storage/innobase/xtrabackup/src/backup_copy.cc:2577]: I think the commit message doesn't talk about this change. Why do we use it? it is escaping the new line characters?
Can you please update commit message? and a comment here too

**Review by satya-bodapati — CHANGES_REQUESTED** [review]

File: storage/innobase/xtrabackup/src/backup_copy.cc (hunk 1 above)
Comment (altmannmarcelo) [reply storage/innobase/xtrabackup/src/backup_copy.cc:2577]: Hi @satya-bodapati .
The fix is about escaping the filepath with single quotes via SQUOTE, however, if the file name has quotes in it we can still break out the command and inject code. escape_string_for_mysql is meant to scape special characters (include quotes) within the file name.

I will adjust the 2.4 PR and re-import it to 8.0

Comment (satya-bodapati) [reply storage/innobase/xtrabackup/src/backup_copy.cc:2577]: Thanks marce, please rerequest review after PR update

**Review by satya-bodapati — APPROVED** [review]
Please merge after circle CI result

**Pull request merged**